src/Routers/UsersRouter.js

```diff
@@ -88,94 +68,154 @@ export class UsersRouter extends ClassesRouter {
-              throw new Parse.Error(Parse.Error.OBJECT_NOT_FOUND, 'Your password has expired. Please reset your password.');
           }
-        }
 
-        delete user.password;
+          delete user.password;
 
-        // Remove hidden properties.
-        UsersRouter.removeHiddenProperties(user);
+          // Remove hidden properties.
+          UsersRouter.removeHiddenProperties(user);
 
-        // Sometimes the authData still has null on that keys
-        // https://github.com/parse-community/parse-server/issues/935
-        if (user.authData) {
-          Object.keys(user.authData).forEach((provider) => {
-            if (user.authData[provider] === null) {
-              delete user.authData[provider];
+          // Sometimes the authData still has null on that keys
+          // https://github.com/parse-community/parse-server/issues/935
+          if (user.authData) {
+            Object.keys(user.authData).forEach((provider) => {
+              if (user.authData[provider] === null) {
+                delete user.authData[provider];
+              }
+            });
+            if (Object.keys(user.authData).length == 0) {
+              delete user.authData;
             }
-          });
-          if (Object.keys(user.authData).length == 0) {
-            delete user.authData;
           }
-        }
-        const {
-          sessionData,
-          createSession
-        } = Auth.createSession(req.config, { userId: user.objectId, createdWith: {
-          'action': 'login',
-          'authProvider': 'password'
-        }, installationId: req.info.installationId });
 
-        user.sessionToken = sessionData.sessionToken;
+          return user;
+        });
+    } catch (error) {
+      return error;
+    }
+  }
+
+  handleMe(req) {
+    if (!req.info || !req.info.sessionToken) {
+      throw new Parse.Error(Parse.Error.INVALID_SESSION_TOKEN, 'invalid session token');
+    }
+    const sessionToken = req.info.sessionToken;
+    return rest.find(req.config, Auth.master(req.config), '_Session',
+      { sessionToken },
+      { include: 'user' }, req.info.clientSDK)
+      .then((response) => {
+        if (!response.results ||
+          response.results.length == 0 ||
+          !response.results[0].user) {
+          throw new Parse.Error(Parse.Error.INVALID_SESSION_TOKEN, 'invalid session token');
+        } else {
+          const user = response.results[0].user;
+          // Send token back on the login, because SDKs expect that.
+          user.sessionToken = sessionToken;
 
-        req.config.filesController.expandFilesInObject(req.config, user);
+          // Remove hidden properties.
+          UsersRouter.removeHiddenProperties(user);
 
-        return createSession();
-      }).then(() => {
-        return { response: user };
+          return { response: user };
+        }
       });
   }
 
+  handleLogIn(req) {
+    try {
```

Comment:
not a big fan of wrapping in try's why do we need those? the top level promise should *never* throw but return a failed promise. Can we change that?

Reply:
I can update this to remove the try / catch's :)